### PR TITLE
Add multi-file matching, new file creation, and alias support

### PR DESCRIPTION
**Pull request by wesen**
Enhance Obsidian-like navigation in VS Code with new features:

- Multi-file matching: Support for files with the same name across the vault
- New file creation: Ability to create non-existent files in a configurable
  inbox folder
- Alias notation: Support for Obsidian's [[link|alias]] syntax

Implementation improvements:
- Full vault watching and lookup for efficient file indexing
- File cache system for improved performance
- File watcher to keep the cache updated
- Updated extension configuration with inbox folder setting
- New command to create files in the specified inbox folder

Other changes:
- Added ignored files that seem to be standard in vscode extensions (esbuild.js, some settings in .vscode)

Disclaimer: I'm new to VSCode extension development (not to development in general though). Most of the code here is from claude/gpt-o1.

### CODE REVIEW

File: README.md
Comment (wesen) [inline README.md]: I should probably be less aggressive in the rewrite.

**Review by geoffolson — CHANGES_REQUESTED** [review]
There are too many changes in a single PR. Can you break this down into multiple PRs with each one focused on a single issue?

Retrospectively, I should have added a CONTRIBUTING.md file to this project. I wasn't expecting to get a contribution so soon.

File: README.md
Comment (wesen) [reply README.md]: of course. Would it make sense to open with a PR that adds the files I think were missing (esbuild.js, some of the .vscode stuff)?

**Comment by wesen** [comment]
of course. Would it make sense to open with a PR that adds the files I think were missing (esbuild.js, some of the .vscode stuff)?

**Comment by geoffolson** [comment]
> of course. Would it make sense to open with a PR that adds the files I think were missing (esbuild.js, some of the .vscode stuff)?

I'm pushing a change to fix the missing `esbuild.js`. I have `*.js` in the `.gitignore` so it was left out by mistake. I'm switching the file to `*.mjs`. I prefer to stick with es modules if possible in the source code.

The `.vscode` directory is also in the .gitignore. Some files from `.vscode` probably should be committed. If you make a PR for the .vscode directory could you also add the exceptions to the `.gitignore` like this for example so the file will be tracked.
```
!.vscode/settings.json
```

File: src/fileCache.ts
Comment (geoffolson) [inline src/fileCache.ts:18]: We don't need this function. The vscode api provides a [findFiles](https://vscode-api.js.org/modules/vscode.workspace.html#findFiles) method that can handle this

Comment (wesen) [reply src/fileCache.ts:18]: perfect!

**Comment by wesen** [comment]
sounds good, i'll wait until your merge and will split my features up into smaller chunks. thanks for having a look! and indeed it's funny that you created it just as i wanted to have a plugin like that :]

**Comment by geoffolson** [comment]
link aliases has been added and I've switched to using `findfiles` so files nested in folders are found now. I don't know if I want to add the inbox feature. There is still no cache mechanism. I don't know how I feel about this. I personally don't see any noticeable performance issues with the current implementation, but that doesn't mean larger vaults wouldn't see the benefits of a cache.

**Comment by geoffolson** [comment]
As of version 0.0.5 the bullet points in this PR are already implemented or out of scope of this extension at the moment. You can raise an issue if you have a feature request or issue.

**Comment by wesen** [comment]
thanks! I've been swamped with work and didn't find time to get back to this.

**Comment by wesen** [comment]
I am fairly sure findFiles has some cache behind the scenes too.

**Pull request closed without merging**
